src/LinearAlgebra/COO_Matrix.hpp

```diff
@@ -726,37 +798,76 @@ inline void COO_Matrix<ScalarT, Intdx>::sortSparseCOO(std::vector<Intdx> &rows,
     }
 }
 
+/**
+ * @brief Constructor for COO Matrix with given cooridnates and values
+ * 
+ * @tparam ScalarT
+ * @tparam Intdx
+ * 
+ * @param[in] r row indices
+ * @param[in] c column indices
+ * @param[in] v values
+ * @param[in] m number of rows
+ * @param[in] n number of columns
+ * 
+ * @pre r.size() == c.size() == v.size()
+ * @pre r,c,v represent an array in COO format
+ * 
+ * @post COO_Matrix is created with given coordinates and values
+ */
+
 template <class ScalarT, typename Intdx>
 inline COO_Matrix<ScalarT, Intdx>::COO_Matrix(std::vector<Intdx> r, std::vector<Intdx> c, std::vector<ScalarT> v, Intdx m, Intdx n)
 {
     this->values_ = v;
-    this->row_indexes_ = r;
-    this->column_indexes_ = c;
+    this->row_indices_ = r;
+    this->column_indices_ = c;
     this->rows_size_ = m;
     this->columns_size_ = n;
-    this->sorted_ = false;
+    this->sorted_ = false; //Any new entry can assume that the matrix is sorted when being added.
 }
 
+/**
+ * @brief Constructor for empty COO Matrix of a given size
+ * 
+ * @tparam ScalarT
+ * @tparam Intdx
+ * 
+ * @param[in] m number of rows
+ * @param[in] n number of columns
+ * 
+ * @post empty COO Matrix is created with given size
+ */
+
 template <class ScalarT, typename Intdx>
 inline COO_Matrix<ScalarT, Intdx>::COO_Matrix(Intdx m, Intdx n)
 {
     this->rows_size_ = m;
     this->columns_size_ = n;
     this->values_ = std::vector<ScalarT>();
-    this->row_indexes_ = std::vector<Intdx>();
-    this->column_indexes_ = std::vector<Intdx>();
-    this->sorted_ = false;
+    this->row_indices_ = std::vector<Intdx>();
+    this->column_indices_ = std::vector<Intdx>();
+    this->sorted_ = true; //Any new entry can assume that the matrix is sorted when being added.
 }
```

Comment:
It is hard to understand what this comment mean here. Did you mean to say "an empty matrix is sorted matrix"?

Reply:
Please clarify that in the comment.